rbd/diff_iterate_by_id.go

```diff
@@ -0,0 +1,136 @@
+//go:build ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#undef _GNU_SOURCE
+#include <errno.h>
+#include <stdlib.h>
+#include <rbd/librbd.h>
+
+#ifdef LIBRBD_SUPPORTS_DIFF_ITERATE3
+#define _RBD_DIFF_ITERATE_FLAG_INCLUDE_PARENT RBD_DIFF_ITERATE_FLAG_INCLUDE_PARENT
+#define _RBD_DIFF_ITERATE_FLAG_WHOLE_OBJECT RBD_DIFF_ITERATE_FLAG_WHOLE_OBJECT
+#define _RBD_DIFF_ITERATE_FLAGS_SUPPORTED 1
+#else
+#define _RBD_DIFF_ITERATE_FLAG_INCLUDE_PARENT 0
+#define _RBD_DIFF_ITERATE_FLAG_WHOLE_OBJECT 0
+#define _RBD_DIFF_ITERATE_FLAGS_SUPPORTED 0
+#endif
```

Comment:
Thanks @phlogistonjohn ! 

This approach works.
Updated the pr
Please take a look